### PR TITLE
Add event comment #3208

### PR DESCRIPTION
**Pull request by innayashna**
I will add this feature (Case comment is added -> send notification to the Organizer about the comment addition with the link to the comment) in the next pull request

### CODE REVIEW

File: core/src/main/java/greencity/controller/EventCommentController.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+     * @param eventId id of {@link greencity.dto.event.EventVO} to add comment to.
+     * @param request - dto for {@link greencity.dto.eventcomment.EventCommentVO}
+     *                entity.
+     * @return dto {@link greencity.dto.eventcomment.AddEventCommentDtoResponse}
+     */
+    @ApiOperation(value = "Add comment.")
+    @ResponseStatus(value = HttpStatus.CREATED)
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = HttpStatuses.CREATED, response = AddEventCommentDtoRequest.class),
+        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
```
Comment (danylohlynsky) [inline core/src/main/java/greencity/controller/EventCommentController.java]: Is it possible to receive this response?

File: core/src/main/java/greencity/controller/EventCommentController.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     *
+     * @param eventId to specify {@link greencity.dto.event.EventVO}
+     * @return amount of comments
+     */
+    @ApiOperation(value = "Count comments.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND),
+    })
+    @GetMapping("/count/comments/{eventId}")
```
Comment (danylohlynsky) [inline core/src/main/java/greencity/controller/EventCommentController.java]: I think you don't need to duplicate word "comments"
Now this controller looks like "events/comments/count/comments/id"

File: core/src/test/java/greencity/controller/EventCommentControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    @BeforeEach
+    void setup() {
+        this.mockMvc = MockMvcBuilders.standaloneSetup(eventCommentController)
+            .setCustomArgumentResolvers(new PageableHandlerMethodArgumentResolver(),
+                new UserArgumentResolver(userService, modelMapper))
+            .build();
+    }
+
+    @Test
+    void save() throws Exception {
```
Comment (danylohlynsky) [inline core/src/test/java/greencity/controller/EventCommentControllerTest.java]: You can use @SneakyThrows annotation here and in the rest of the tests, no need to have this exception

File: dao/src/main/resources/db/changelog/logs/ch-add-event-comment-Yashna.xml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="event_id" type="BIGINT">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+    </changeSet>
+
+    <changeSet id="Yashna-2" author="Inna Yashna">
```
Comment (danylohlynsky) [inline dao/src/main/resources/db/changelog/logs/ch-add-event-comment-Yashna.xml]: I think you can unite all these 3 changeSets into 1 as they work with the same table

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+     *
+     * @param eventId                   id of {@link greencity.entity.event.Event}
+     *                                  to which we save comment.
+     * @param addEventCommentDtoRequest dto with
+     *                                  {@link greencity.entity.event.EventComment}
+     *                                  text, parentCommentId.
+     * @param userVO                    {@link User} that saves the comment.
+     * @return {@link AddEventCommentDtoResponse} instance.
+     */
+
```
Comment (danylohlynsky) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java]: Remove the empty line please

File: service/src/main/java/greencity/service/EventCommentServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+     * Method to get all active comments to {@link Event} specified by eventId.
+     *
+     * @param pageable page of event.
+     * @param eventId  specifies {@link Event} to which we search for comments
+     * @return all active comments to certain event specified by eventId.
+     */
+    @Override
+    public PageableDto<EventCommentDto> getAllActiveComments(Pageable pageable, UserVO userVO, Long eventId) {
+        Optional<Event> event = eventRepo.findById(eventId);
```
Comment (danylohlynsky) [inline service/src/main/java/greencity/service/EventCommentServiceImpl.java:76]:
```suggestion
       Event event = eventRepo.findById(eventId).orElseThrow(() -> new NotFoundException(ErrorMessage.EVENT_NOT_FOUND_BY_ID + eventId));
```
With this you can remove next lines

**Review by danylohlynsky — APPROVED** [review]

**Review by OlegVat — APPROVED** [review]

**Review by NataliaHlazova — APPROVED** [review]

File: core/src/main/java/greencity/controller/EventCommentController.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    /**
+     * Method to count active comments to certain
+     * {@link greencity.dto.event.EventVO}.
+     *
+     * @param eventId to specify {@link greencity.dto.event.EventVO}
+     * @return amount of comments
+     */
+    @ApiOperation(value = "Count comments.")
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/EventCommentController.java:71]: Please add more info

File: core/src/test/java/greencity/controller/EventCommentControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ContextConfiguration
+@Import(SecurityConfig.class)
+class EventCommentControllerTest {
+    private static final String eventCommentControllerLink = "/events/comments";
```
Comment (RomaMocherniuk) [inline core/src/test/java/greencity/controller/EventCommentControllerTest.java]: Please use Java convention for naming

File: dao/src/main/java/greencity/entity/event/EventComment.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+import javax.validation.constraints.Size;
+import java.time.LocalDateTime;
+
+@Entity
+@Table(name = "events_comment")
+@Getter
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/entity/event/EventComment.java]: You can use @Data annotation
https://www.educative.io/answers/what-is-the-data-annotation-in-lombok

**Pull request merged**